File: src/Atlasd/Battlenet/AccountKeyValue.cs
namespace Atlasd.Battlenet
{
    class AccountKeyValue
    {
        public enum ReadLevel
        {
            Any,
            Owner,
            Internal,
        };
        public enum WriteLevel
        {
            Any,
            Owner,
            Internal,
            ReadOnly,
        };

        public string Key { get; private set; }
        public ReadLevel Readable { get; private set; }
        public dynamic Value;
        public WriteLevel Writable { get; private set; }

        public AccountKeyValue(string key, dynamic value, ReadLevel readable, WriteLevel writable)
        {
            Key = key;
            Readable = readable;
            Value = value;
            Writable = writable;
        }
    }
}

File: src/AtlasV/Battlenet/BNET/AccountKeyValue.vb

Namespace AtlasV.Battlenet
    Class AccountKeyValue
        Public Enum ReadLevel
            Any
            Owner
            Internal
        End Enum

        Public Enum WriteLevel
            Any
            Owner
            Internal
            [ReadOnly]
        End Enum

        Public Property Key As String
        Public Property Readable As ReadLevel
        Public Value As Object
        Public Property Writable As WriteLevel

        Public Sub New(ByVal varKey As String,
                       ByVal varValue As Object,
                       ByVal varReadable As ReadLevel,
                       ByVal varWritable As WriteLevel)
            Key = varKey
            Value = varValue
            Readable = varReadable
            Writable = varWritable
        End Sub
    End Class
End Namespace
